src/Service.GraphQLBuilder/GraphQLUtils.cs

```diff
@@ -387,6 +388,45 @@ private static string GenerateDataSourceNameKeyFromPath(IMiddlewareContext conte
             return $"{context.Path.ToList()[0]}";
         }
 
+        /// <summary>
+        /// Helper method to determine whether a field is a column or complex (relationship) field based on its syntax kind.
+        /// If the SyntaxKind for the field is not ObjectValue and ListValue, it implies we are dealing with a column/scalar field which
+        /// has an IntValue, FloatValue, StringValue, BooleanValue or an EnumValue.
+        /// </summary>
+        /// <param name="fieldSyntaxKind">SyntaxKind of the field.</param>
+        /// <returns>true if the field is a scalar field, else false.</returns>
+        public static bool IsScalarField(SyntaxKind fieldSyntaxKind)
+        {
+            return fieldSyntaxKind is SyntaxKind.IntValue || fieldSyntaxKind is SyntaxKind.FloatValue ||
+                fieldSyntaxKind is SyntaxKind.StringValue || fieldSyntaxKind is SyntaxKind.BooleanValue ||
```

Comment:
Should we check for `SyntaxKind.NullValue` here to call it a scalar field?

Reply:
Nope, NullValue can also be for a relationship field since relationship fields are optional.